Migrate all Python3 Scripts to uv

MickLesk opened the pull request:
## ✍️ Description  


| Script Name             | Bash Call                                                                                                                        | Install Done | Update Done | Install working? | Update/Migration working? | Error Msg |
|:----------------------- |:---------------------------------------------------------------------------------------------------------------------------------|:------------:|:-----------:|:----------------:|:------------------------:|:---------:|
| archivebox              | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/archivebox.sh)"              | ✅           | ❌          | ❌               |                        |           |
| babybuddy               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/babybuddy.sh)"               | ✅           | ❌          |                  |                        |           |
| bazarr                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/bazarr.sh)"                  | ✅           | ❌          | ✅               |                        |           |
| calibre-web             | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/calibre-web.sh)"             | ✅           | ❌          |                  |                        |           |
| changedetection         | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/changedetection.sh)"         | ✅           | ❌          |                  |                        |           |
| crafty-controller       | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/crafty-controller.sh)"       | ✅           | ❌          |                  |                        |           |
| deluge                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/deluge.sh)"                  | ✅           | ❌          |                  |                        |           |
| documenso               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/documenso.sh)"               | ✅           | ❌          |                  |                        |           |
| grist                   | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/grist.sh)"                   | ✅           | ❌          | ✅               |                        |           |
| homer                   | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/homer.sh)"                   | ✅           | ❌          |                  |                        |           |
| immich                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/immich.sh)"                  | ✅           | ❌          |                  |                        |           |
| kometa                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/kometa.sh)"                  | ✅           | ❌          |                  |                        |           |
| lazylibrarian           | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/lazylibrarian.sh)"           | ✅           | ❌          |                  |                        |           |
| medusa                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/medusa.sh)"                  | ✅           | ❌          |                  |                        |           |
| motioneye               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/motioneye.sh)"               | ✅           | ❌          |                  |                        |           |
| mylar3                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/mylar3.sh)"                  | ✅           | ❌          |                  |                        |           |
| netbox                  | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/netbox.sh)"                  | ✅           | ❌          |                  |                        |           |
| nginxproxymanager       | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/nginxproxymanager.sh)"       | ✅           | ❌          |                  |                        |           |
| octoprint               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/octoprint.sh)"               | ✅           | ❌          |                  |                        |           |
| openwebui               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/openwebui.sh)"               | ✅           | ❌          |                  |                        |           |
| paperless-ai            | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/paperless-ai.sh)"            | ✅           | ❌          |                  |                        |           |
| paperless-ngx           | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/paperless-ngx.sh)"           | ✅           | ❌          |                  |                        |           |
| pialert                 | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/pialert.sh)"                 | ✅           | ❌          |                  |                        |           |
| radicale                | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/radicale.sh)"                | ✅           | ❌          |                  |                        |           |
| sabnzbd                 | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/sabnzbd.sh)"                 | ✅           | ❌          |                  |                        |           |
| searxng                 | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/searxng.sh)"                 | ✅           | ❌          |                  |                        |           |
| skskd                   | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/skskd.sh)"                   | ✅           | ❌          |                  |                        |           |
| spoolman                | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/spoolman.sh)"                | ✅           | ❌          | ✅               |                        |           |
| stirling-pdf            | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/stirling-pdf.sh)"            | ✅           | ❌          | ✅               |                        |           |
| tandoor                 | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/tandoor.sh)"                 | ✅           | ❌          | ❌               |                        | No `project` table found in: `/opt/tandoor/pyproject.toml` ⠹ [ERROR] in line 63: exit code 0: while executing command $STD uv run python version.py |
| tasmocompiler           | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/tasmocompiler.sh)"           | ✅           | ❌          | ✅               |                        |           |
| tautulli                | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/tautulli.sh)"                | ✅           | ❌          | ✅               |                        |           |
| unmanic                 | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/unmanic.sh)"                 | ✅           | ❌          | ✅               |                        |           |
| web-check               | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/web-check.sh)"               | ✅           | ❌          | ✅               |                        |           |
| wger                    | bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/python_to_uv/ct/wger.sh)"                    | ✅           | ❌          | ❌               |                        | line 48: /home/wger/.venv/bin/uv: No such file or directory while executing command $STD /home/wger/.venv/bin/uv run python manage.py collectstatic |



## 🔗 Related PR / Issue  
Link: #


## ✅ Prerequisites  (**X** in brackets) 

- [ ] **Self-review completed** – Code follows project standards.  
- [ ] **Tested thoroughly** – Changes work as expected.  
- [ ] **No security risks** – No hardcoded secrets, unnecessary privilege escalations, or permission issues.  

---

## 🛠️ Type of Change (**X** in brackets)  

- [ ] 🐞 **Bug fix** – Resolves an issue without breaking functionality.  
- [ ] ✨ **New feature** – Adds new, non-breaking functionality.  
- [ ] 💥 **Breaking change** – Alters existing functionality in a way that may require updates.  
- [ ] 🆕 **New script** – A fully functional and tested script or script set.  
- [ ] 🌍 **Website update** – Changes to website-related JSON files or metadata.  
- [ ] 🔧 **Refactoring / Code Cleanup** – Improves readability or maintainability without changing functionality.  
- [ ] 📝 **Documentation update** – Changes to `README`, `AppName.md`, `CONTRIBUTING.md`, or other docs.

MickLesk committed:
Update build.func

MickLesk committed:
fix testing preparation

MickLesk committed:
migrations

MickLesk committed:
PYTHON_VERSION="3.12" setup_uv

MickLesk committed:
more uv migrations

MickLesk committed:
next migrations

MickLesk committed:
more uv migrations

MickLesk committed:
more uv mig

MickLesk committed:
Update netbox-install.sh

MickLesk committed:
more uv migs

MickLesk committed:
more migrations

MickLesk committed:
more uv migrations

MickLesk committed:
more uv migs

MickLesk committed:
more uv migs

MickLesk committed:
Update slskd-install.sh

MickLesk committed:
Update nginxproxymanager-install.sh

MickLesk committed:
more uv

MickLesk committed:
AdventureLog: Full UV Mig

MickLesk committed:
Archivebox: UV Migration

MickLesk committed:
optimize uv python install

MickLesk committed:
fix python install for uv

MickLesk committed:
various improvements on uv setup

MickLesk committed:
fixes

MickLesk committed:
Update archivebox-install.sh

MickLesk committed:
change paths for testing

MickLesk committed:
Update adventurelog-install.sh

MickLesk committed:
Update archivebox-install.sh

MickLesk committed:
Update tools.func

MickLesk committed:
Update adventurelog-install.sh

MickLesk committed:
Update archivebox-install.sh

MickLesk committed:
Update adventurelog-install.sh

MickLesk committed:
fix uv install

MickLesk committed:
some fixes with venv

MickLesk committed:
fix pip for adventurelog

MickLesk committed:
some fixes

MickLesk committed:
remove "This will create an ... LXC"

MickLesk committed:
Update archivebox-install.sh

MickLesk committed:
harmonize .venv

michelroegl-brunner committed:
Change from python3.12 to python3 for venv

michelroegl-brunner committed:
Change user

MickLesk committed:
fixes

MickLesk committed:
add build-essential to deluge

MickLesk committed:
some fixes

MickLesk committed:
fix tasmocompiler

MickLesk committed:
more uv migs

MickLesk committed:
testing

MickLesk committed:
more migs

MickLesk committed:
more

MickLesk committed:
fix deluge

MickLesk committed:
fixes

MickLesk committed:
Update wger-install.sh

MickLesk committed:
fixes

MickLesk committed:
test sabnzd

MickLesk committed:
test

MickLesk committed:
Update tools.func

tremor021 committed:
MeTube to Python 3.13

vhsdream commented on install/immich-install.sh:
```diff
@@ -304,25 +304,25 @@ cp LICENSE "$APP_DIR"
 cd "$SRC_DIR"/machine-learning
-$STD python3 -m venv "$ML_DIR/ml-venv"
+$STD uv venv "$ML_DIR/.venv"
+$ML_DIR/.venv/bin/uv pip install --upgrade uv
+$ML_DIR/.venv/bin/uv sync --no-cache
+cd "$SRC_DIR"/machine-learning
+$STD uv venv "$ML_DIR/.venv"
+$ML_DIR/.venv/bin/uv pip install --upgrade uv
```
I'm a little confused here. If uv is installed by default then this isn't really necessary. It looks like we're using uv to create a virtualenv, then using the uv inside that virtualenv (which isn't actually there by the way) to install/upgrade uv??

MickLesk replied:
You use build in Python to Install UV and this uv use the build in Python, so #555 can be occured Sometimes

MickLesk replied:
Ah i See you are in my pr. Immich is mot ready, only 6 Scripts are done atm

vhsdream reviewed (changes requested):
For line 307-323 (immich-install.sh), this works:

```bash
cd "$SRC_DIR"/machine-learning
export VIRTUAL_ENV="${ML_DIR}/ml-venv"
$STD uv venv "$VIRTUAL_ENV"
if [[ -f ~/.openvino ]]; then
  msg_info "Installing HW-accelerated machine-learning"
  uv -q sync --extra openvino --no-cache --active
  patchelf --clear-execstack "$ML_DIR"/ml-venv/lib/python3.11/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.cpython-311-x86_64-linux-gnu.so
  msg_ok "Installed HW-accelerated machine-learning"
else
  msg_info "Installing machine-learning"
  uv -q sync --extra cpu --no-cache --active
  msg_ok "Installed machine-learning"
fi
```
I've also found that the output of uv is not suppressed when using $STD, but using `-q` seems to do it, with the drawback that it will also suppress output during verbose operation.

Also, `python3-dev` (line 31) is required, otherwise the machine-learning build will fail

vhsdream committed:
Immich UV test (#5032)

- Simplifies UV use in Immich install & update
- Uses Python 3.11 (No official support for 3.12 yet)
- Restores python3-dev package

MickLesk committed:
Update motioneye-install.sh

MickLesk committed:
fixes

MickLesk committed:
fixes

MickLesk commented:
Current Migrations:
#5318 
#5320
#5411

MickLesk committed:
Update prometheus-pve-exporter.sh

MickLesk committed:
fixes

MickLesk committed:
Update spoolman-install.sh

MickLesk committed:
update tools.func for migration

MickLesk committed:
Update spoolman-install.sh

MickLesk committed:
Update spoolman-install.sh
